main.py

```diff
@@ -126,16 +250,26 @@ def evaluate_and_write(args, model, tasks, splits_to_write):
     log.info("Writing results for split 'val' to %s", results_tsv)
     evaluate.write_results(val_results, results_tsv, run_name=run_name)
 
-
-def main(cl_arguments):
-    ''' Train or load a model. Evaluate on some tasks. '''
-    cl_args = handle_arguments(cl_arguments)
-    args = config.params_from_file(cl_args.config_file, cl_args.overrides)
-
-    # Raise error if obsolete arg names are present
-    check_arg_name(args)
-
-    # Logistics #
+def setup(args, cl_args):
+    """
+    Sets up email hook, cuda settings, and various initial sanity checks such as 
+    not mixing tasks with minimizing and maximizing validation set objective functions.
```

Comment:
Can we put these sanity checks with the other sanity checks in the earlier method?